core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -639,4 +640,25 @@ impl ZksNamespace {
             storage_proof,
         })
     }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn get_batch_pubdata_impl(
+        &self,
+        l1_batch_number: L1BatchNumber,
+    ) -> Result<Option<Vec<u8>>, Web3Error> {
+        const METHOD_NAME: &str = "get_batch_pubdata";
+
+        let method_latency = API_METRICS.start_call(METHOD_NAME);
+        self.state.start_info.ensure_not_pruned(l1_batch_number)?;
+        let mut storage = self.access_storage(METHOD_NAME).await?;
+        let l1_batch_with_metadata = storage
+            .blocks_dal()
+            .get_l1_batch_metadata(l1_batch_number)
+            .await
+            .unwrap_or(None);
```

Comment:
This doesn't look right; DB-level errors should not be swallowed like this, but rather should be converted to internal server errors. See other methods in this module for the reference.